Homework Solutions/Week 1/Lecture 1/single-element-in-a-sorted-array.cpp

```diff
@@ -0,0 +1,102 @@
+// Ali Emre Kaya
+// Question Link: https://leetcode.com/problems/single-element-in-a-sorted-array/
+
+/*
+The problem requires finding the single element in a sorted array where all other elements appear exactly twice.
+We can find it with visiting all the element (brute force approach) until see sngle eleemnt, it's time complexity will be O(n) and will not pass.
+Instead of brute force approach, we can use a binary search approach to solve this efficiently in O(log n) time.
+ 
+Key Idea: If we look at the pairs in the array, the single element disrupts the pairing.
+We exploit this by checking pairs and using binary search to find the single element.
+*/
+
```

Comment:
please provide time and (optionally) space complexity of the solution in a seperated line. Like
```c++
// Time complexity: O(log n)
// Space complexity: O(1)
```